### PR TITLE
Updated assets restore to preserve file permissions

### PR DESCRIPTION
**Pull request by NZI**

### CODE REVIEW

**Comment by NZI** [comment]
https://github.com/silverstripe/sspak/issues/65

File: src/Webroot.php [hunk 1]
```diff
@@ -43,12 +43,12 @@ public function sniff() {
 	 * Execute a command on the relevant server, using the given sudo option
 	 * @param  string $command Shell command, either a fully escaped string or an array
 	 */
-	public function execSudo($command) {
+	public function execSudo($command, $options = array()) {
```
Comment (andrewandante) [inline src/Webroot.php]: Please update the docblock to reflect the additional parameter.

File: src/Webroot.php [hunk 2]
```diff
@@ -147,16 +147,20 @@ public function putassets($sspak) {
 		$assetsOldPath = $assetsPath . '.old';
 		$assetsParentArg = escapeshellarg(dirname($assetsPath));
 
+		// Remove assets.old
```
Comment (andrewandante) [inline src/Webroot.php]: Can you explain the reasoning behind moving this to before the restore? This is functional change and seems out of scope of the PR - and I'm happy to take a look at it separately, it seems to have little to do with fixing the issue where `$options` aren't passed through from `sudo`.

Otherwise it seems to me that the `mv` command will replace the `assets.old` dir, so we don't need to wipe it.

File: src/Webroot.php (hunk 1 above)
Comment (andrewandante) [reply src/Webroot.php]: @dhensby are we consciously not passing `$options` through via sudo for any reason, security or otherwise?

File: src/Webroot.php [hunk 3]
```diff
@@ -147,16 +147,20 @@ public function putassets($sspak) {
...
+		}
 
 		// Extract assets
 		$stream = $sspak->readStreamForFile('assets.tar.gz');
-		$this->exec("tar xzf - -C {$assetsParentArg}", array('inputStream' => $stream));
+		$this->execSudo("tar xzf - -C {$assetsParentArg}", array('inputStream' => $stream));
 		fclose($stream);
-
-		// Remove assets.old
-		$this->exec("if [ -d {$assetsOldPath} ]; then rm -rf {$assetsOldPath}; fi");
```
Comment (andrewandante) [inline src/Webroot.php]: This is also outside the scope of the PR - I'd include it in the above one, with your reasoning. I imagine it's something like "keeping a backup is better then saving space", but that's a different conversation 😄

File: src/Webroot.php (hunk 1 above)
Comment (dhensby) [reply src/Webroot.php]: 🤷‍♂️ I'm not sure :/

File: src/Webroot.php (hunk 3 above)
Comment (NZI) [reply src/Webroot.php]: I must've misinterpreted what assets.old was for when I first was using sspak... Yea i'll put these back the other way, no reasoning to the reorder

File: src/Webroot.php [hunk 4]
```diff
@@ -148,15 +154,19 @@ public function putassets($sspak) {
 		$assetsParentArg = escapeshellarg(dirname($assetsPath));
 
 		// Move existing assets to assets.old
-		$this->exec("if [ -d {$assetsPath} ]; then mv {$assetsPath} {$assetsOldPath}; fi");
+		if (file_exists($assetsPath)) {
```
Comment (dhensby) [inline src/Webroot.php]: Why have we moved the `file_exists` check into PHP and away from bash, which would execute under the sudo user (which is probably important?)

Comment (NZI) [reply src/Webroot.php]: I will push a version without file_exists shortly. Just testing it now.

**Comment by dhensby** [comment]
@andrewandante happy?

**Comment by andrewandante** [comment]
Yep yep, merge away

**Comment by dhensby** [comment]
thanks @nzi!
